pkg/hive/hive.go

```diff
@@ -399,9 +394,16 @@ func (h *Hive) PrintDotGraph() {
 	}
 }
 
+// SetEnvPrefix globally sets the enviroment prefix to use with the hive package.
+// This should be used early in initialization and only once as it affects all hives
+// in the program.
+func SetEnvPrefix(prefix string) {
```

Comment:
I appreciate this isn't a change with this PR, but it seems unusual that `SetEnvPrefix` has different behavior to Viper's `SetEnvPrefix` [equivalent](https://github.com/spf13/viper#working-with-environment-variables). In Viper, the argument is automatically uppercased, and an underscore is automatically inserted. 

With this implementation, you'd need to call `hive.SetEnvPrefix("FOO_")` and not `hive.SetEnvPrefix("foo")`, as the latter would expect environment variables like `fooSOME_FLAG` for `--some-flag`.

As the previous implementation of `SetEnvPrefix` didn't work anyway, we can be sure we won't break existing implementations if we change the behaviour now, so maybe now would be a good time to change it?

Reply:
Good idea! I'll make it work the same.